common/futures-semaphore/src/lib.rs

```diff
@@ -1,7 +1,7 @@
 // Copyright (c) The Libra Core Contributors
 // SPDX-License-Identifier: Apache-2.0
 
-#![forbid(unsafe_code)]
+// #![forbid(unsafe_code)]
```

Comment:
Please uncomment this, as it looks like there isn't any unsafe code we should still forbid it

Reply:
Ahh woops, thanks for catching that. I was messing around with `mem::transmute` to fix the lifetime issue before I figured out the current approach